src/internal_events/exec.rs

```diff
@@ -38,51 +44,135 @@ impl InternalEvent for ExecEventsReceived<'_> {
 }
 
 #[derive(Debug)]
-pub struct ExecFailed<'a> {
+pub struct ExecFailedError<'a> {
     pub command: &'a str,
     pub error: std::io::Error,
 }
 
-impl InternalEvent for ExecFailed<'_> {
+impl InternalEvent for ExecFailedError<'_> {
     fn emit_logs(&self) {
         error!(
             message = "Unable to exec.",
             command = %self.command,
             error = ?self.error,
+            error_type = "failed",
```

Comment:
I think we need something a little more specific than just `failed` here and below, though I'm not seeing another error type that we are already using that would be appropriate. AFAICT the `exec` source is the only one using the bare word `failed`. Maybe `source_failed` although that is pretty generic.

Reply:
I've a small preference for `command_failed` considering it the executed command that failed 🤔